### PR TITLE
Expand docs for hook environments

### LINKED ISSUE #239: Clarify meaning of "hook environments"
In sections [pre-commit install](https://pre-commit.com/#pre-commit-install) and [pre-commit install-hooks](https://pre-commit.com/#pre-commit-install-hooks) are mentioned "hook environments"... Not sure what this means... Could you clarify?

Could not test what result it bring as `pre-commit install --install-hooks` only returns `pre-commit installed at .git/hooks/pre-commit` as message...

### PR DESCRIPTION
**Pull request by rpdelaney**
Fixes #239 and #225

### CODE REVIEW

File: index.mako [hunk 1]
```diff
@@ -758,25 +758,36 @@ Install the pre-commit script.
...
 ## pre-commit install-hooks [options] [](#pre-commit-install-hooks)
 
-Install hook environments for all environments in the config file.  You may
-find `pre-commit install --install-hooks` more useful.
+Install all missing environments for the available hooks. Unless this command or
+`install --install-hooks` is executed, each hook's environment is created the
+first time the hook is called.
+
+Each hook is initialized in a separate environment appropriate to the language
+the hook is written in. See [supported languages](#Supported-languages).
```
Comment (asottile) [inline index.mako]: anchors will be lower-cased `#supported-languages`

File: index.mako [hunk 2]
```diff
@@ -758,25 +758,36 @@ Install the pre-commit script.
...
+  (rather than when they are first executed). See [`pre-commit
+  install-hooks`](#pre-commit-install-hooks).
 - `-t {pre-commit,pre-push,prepare-commit-msg,commit-msg}`,
   `--hook-type {pre-commit,pre-push,prepare-commit-msg,commit-msg}`:
-  which hook type to install.
-- `--allow-missing-config`: whether to allow the installed hook scripts to
-  permit a missing configuration file.
+  Specify which hook type to install.
+- `--allow-missing-config`: Hook scripts will permit a missing configuration
+  file.
```
Comment (asottile) [inline index.mako]: would you like to backport these changes to the help text? (iirc these are grafted directly from there)

Comment (rpdelaney) [reply index.mako]: Sure, can you point me in that direction? This was my first encounter with `.mako` files

Comment (asottile) [reply index.mako]: ah I mean in the code :) https://github.com/pre-commit/pre-commit/blob/master/pre_commit/main.py

Comment (rpdelaney) [reply index.mako]: Oh, sure

**Pull request merged**